demisto_sdk/commands/common/update_id_set.py

```diff
@@ -1012,20 +1030,58 @@ def get_depends_on(data_dict):
     return depends_on_list, command_to_integration
 
 
-def process_integration(file_path: str, print_logs: bool) -> list:
+def should_skip_item_by_mp(file_path: str, marketplace: str):
+    """
+    Checks if the item given (as path) should be part of the current generated id set.
+     The checks are in this order:
+     1. If we are in a xsoar run, then all packs should be uploaded.
+     2. Otherwise, this is not a xsoar run, we check if the given item has the right marketplaces under the 'marketplaces' in the item's file,
+     3. Otherwise, we check if the item is inside a pack that it's pack metadata 'marketplaces' field does not include
+     the current marketplace we are creating this id set for.
+
+    Args:
+        file_path: path to content item
+        marketplace: the marketplace this current generated id set is for
+
+    Returns: True if should be skipped, else False
+
+    """
+    # first check, if we are using this id set for xsoar MP
+    if marketplace == MarketplaceVersions.XSOAR.value:
+        return False
+
+    # second check, check field 'marketplaces' in the item's file
+    file_type = Path(file_path).suffix
+    try:
+        item_data = get_file(file_path, file_type)
+        if marketplace not in item_data.get('marketplaces', []):
```

Comment:
if the field doesn't exist in the file, continue to the next check (the pack metadata).
if the field doesn't exist there too, consider it as XSOAR only